paddle/framework/prune.cc

```diff
@@ -99,11 +102,23 @@ void prune_impl(const ProgramDesc& input, ProgramDesc* output, int block_id) {
       *op_field->Add() = input.blocks(block_id).ops(i);
     }
   }
+  if (is_test) {
+    for (auto& op_desc : *op_field) {
+      if (op_desc.type() == kDropOutOpType ||
+          op_desc.type() == kBatchNormOpType) {
+        for (auto& attr : *op_desc.mutable_attrs()) {
+          if (attr.name() == "is_test") {
+            attr.set_b(true);
+          }
+        }
+      }
+    }
+  }
 }
 
 // TODO(fengjiayi): Prune() could be inplaced to avoid unnecessary copies
-void Prune(const ProgramDesc& input, ProgramDesc* output) {
-  prune_impl(input, output, 0);
+void Prune(const ProgramDesc& input, ProgramDesc* output, bool is_test) {
+  prune_impl(input, output, 0, is_test);
```

Comment:
I think setting attribute `is_test` true has no relationship with `prune`. We shall add an independent function or module to do this.

Reply:
We now have `prune` method to transform ProgramDesc and generate another ProgramDesc.
Exactly, we will do some optimization in inference stage and handle some special operators. Then, we will generate a inference ProgramDesc.
Maybe we need another method like `prune`, the interface could be:
```
void InferenceOptimize(const ProgramDesc& input, ProgramDesc* output);
```